post_info.php

```diff
@@ -0,0 +1,788 @@
+    }
+    if ($original_choices_id_array[2]) {
+        $bind_variables["choice_id_section"] = $original_choices_id_array[2];
+    }
+    // Use a loop to go through the SQL statement iteratively
+    for ($i = 0; $i < count($sql_original_choices); $i++) {
+        $choices = execute_sql(
+            $sql_original_choices[$i],
+            $bind_variables,
+            "select"
+        );
+        if ($choices) {
+            array_push($choices_array, $choices[0]);
+        } else {
+            array_push($choices_array, null);
+        }
+    }
+
+    /* 4. Create new choices instance */
+    $temp_array = set_new_choices($choices_array);
+    $new_choices_id_array = $temp_array["new_choices_id_array"];
+    $sql_get_new_choices = $temp_array["sql_get_new_choices"];
+
+    /* 5. Set up the new survey_choices instance */
+    $new_survey_choices_id_array = set_new_survey_choices(
+        $choices_array,
+        $new_choices_id_array,
+        $department_name,
+        $course_code,
+        $section_id,
+        $user_id,
+        $sql_get_new_choices
+    );
+
+    /* 6. Last step, set a new survey */
+    $sql_set_new_survey = gen_query_set_new_survey();
+    $bind_variables = array();
+    $bind_variables["dept_survey_choice_id"] = $new_survey_choices_id_array[0];
+    $bind_variables[
+        "course_survey_choice_id"
+    ] = $new_survey_choices_id_array[1];
+    $bind_variables["ta_survey_choice_id"] = $new_survey_choices_id_array[2];
+    $bind_variables["name"] = $data->name;
+    $bind_variables["term"] = $data->term;
+    $bind_variables["default_survey_open"] = $original_survey_info_array[
+        "default_survey_open"
+    ];
+    $bind_variables["default_survey_close"] = $original_survey_info_array[
+        "default_survey_close"
+    ];
+    $status = execute_sql($sql_set_new_survey, $bind_variables, null);
+    if ($status != "success" && $status != null) {
+        $return_data["TYPE"] = "error";
+        $return_data["data"] = $status;
+        echo json_encode($return_data);
+        exit();
+    } else {
+        echo json_encode($return_data);
+        exit();
+    }
+}
+
+/**
+ *  return the according department_name,course_code, section_id
+ *  and the original_choices_id_array
+ * @param level:string "dept", "course", "section"
+ * @param original_survey_info_array:array An associative array that includes
+ *        the information of the original survey
+ * @return array associative array that includes all the necessary information for branching
+ */
+function get_choices_id(
+    $level,
+    $original_survey_info_array,
+    $department_name,
+    $course_code,
+    $section_id
+) {
+    $sql_choice_id = gen_query_get_choices_id($level);
+    $original_choices_id_array = array();
+    for ($i = 0; $i < count($sql_choice_id); $i++) {
+        $bind_variables = array();
+        if ($i == 0 && $original_survey_info_array["dept_survey_choice_id"]) {
```

Comment:
I don't like if statements checking `$i`. Store the data that needs to change based on `$i` in an array and then use `$i` to access that data.